programs/protocol_event/src/lib.rs

```diff
@@ -14,42 +14,73 @@ declare_id!("5qCutonYoeg1aRK31mv4oQYoKdNFMpPaEtDe9nnNQXXf");
 pub mod protocol_event {
     use super::*;
 
+    // Event management instructions
+
     pub fn create_event(
         ctx: Context<CreateEvent>,
         event_info: CreateEventInfo,
     ) -> Result<()> {
-        instructions::create(ctx, event_info)?;
+        instructions::create_event::create(ctx, event_info)?;
         Ok(())
     }
 
     pub fn activate_event(
         ctx: Context<UpdateEvent>,
         _slug: String,
     ) -> Result<()> {
-        instructions::update::update_active_flag(ctx, true)
+        instructions::update_event::update_active_flag(ctx, true)
     }
 
     pub fn deactivate_event(
         ctx: Context<UpdateEvent>,
         _slug: String,
     ) -> Result<()> {
-        instructions::update::update_active_flag(ctx, false)
+        instructions::update_event::update_active_flag(ctx, false)
     }
 
-    pub fn update_participants(
+    pub fn update_event_participants(
         ctx: Context<UpdateEvent>,
         _slug: String,
         participants: Vec<u16>,
     ) -> Result<()> {
-        instructions::update::update_participants(ctx, participants)
+        instructions::update_event::update_participants(ctx, participants)
     }
 
     pub fn update_expected_start_timestamp(
         ctx: Context<UpdateEvent>,
         _slug: String,
         updated_timestamp: i64,
     ) -> Result<()> {
-        instructions::update::updated_expected_start_timestamp(ctx, updated_timestamp)
+        instructions::update_event::updated_expected_start_timestamp(ctx, updated_timestamp)
+    }
+
+    // Participant management instructions
+
+    pub fn create_individual_participant(
+        ctx: Context<CreateParticipant>,
+        code: String,
+        name: String,
+    ) -> Result<()> {
+        instructions::create_participant::create_individual_participant(ctx, code, name)
+    }
+
+    pub fn create_team_participant(
+        ctx: Context<CreateParticipant>,
+        code: String,
+        name: String,
+    ) -> Result<()> {
+        instructions::create_participant::create_team_participant(ctx, code, name)
+    }
```

Comment:
I split these 2 into separate instructions as they most likely need slightly different logic in future.. which is totally the real reason I split them, and not just that i wanted to avoid dealing with enums as input params